packages/tpetra/core/example/BlockCrs/driver.cpp

```diff
@@ -0,0 +1,536 @@
+    RCP<tpetra_crs_graph_type> bcrs_graph;
+    { 
+      RCP<TimeMonitor> timerGlobalGraphConstruction
+        = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("1) GlobalGraphConstruction")));
+      bcrs_graph = rcp(new tpetra_crs_graph_type(row_map, col_map, local_graph_type(colidx, rowptr),
+                                                Teuchos::null));
+    } // end global graph timer
+    
+#if PRINT_VERBOSE
+    bcrs_graph->describe(*out, Teuchos::VERB_EXTREME);
+#endif
+
+    // Create BlockCrsMatrix
+    RCP<tpetra_blockcrs_matrix_type> A_bcrs = Teuchos::rcp(new tpetra_blockcrs_matrix_type(*bcrs_graph, blocksize));
+
+    typedef tpetra_blockcrs_matrix_type::little_block_type block_type;
+    Kokkos::View<block_type*,exec_space> blocks;
+    {
+      RCP<TimeMonitor> timerLocalBlockCrsFill
+        = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("2) LocalBlockCrsFill")));
+
+      // Tpetra BlockCrsMatrix only has high level access functions
+      // To fill this on device, we need an access to the meta data of blocks
+      const auto rowptr_host = Kokkos::create_mirror_view(rowptr);      
+      const auto colidx_host = Kokkos::create_mirror_view(colidx);
+      
+      Kokkos::deep_copy(rowptr_host, rowptr);
+      Kokkos::deep_copy(colidx_host, colidx);
+
+      blocks = Kokkos::View<block_type*,exec_space>("blocks", rowptr_host(num_owned_elements));
+
+      const auto blocks_host = Kokkos::create_mirror_view(blocks);
+      Kokkos::parallel_for
+        (Kokkos::RangePolicy<host_space>(0,num_owned_elements), KOKKOS_LAMBDA(const LO row) {
+          for (LO loc=rowptr_host(row);loc<rowptr_host(row+1);++loc) 
+            blocks_host(loc) = A_bcrs->getLocalBlock(row, colidx(loc));
+        });    
+      Kokkos::deep_copy(blocks, blocks_host);
+
+      Kokkos::parallel_for
+        (num_owned_elements, KOKKOS_LAMBDA(const LO row) {
+          for (LO loc=rowptr(row);loc<rowptr(row+1);++loc) {
+            const GO gid_row = mesh_gids(row);
+            const GO gid_col = mesh_gids(colidx(loc));
+            
+            LO i0, j0, k0, i1, j1, k1;
+            sb.idx_to_ijk(gid_row, i0, j0, k0);
+            sb.idx_to_ijk(gid_col, i1, j1, k1);
+            
+            const LO diff_i = i0 - i1;
+            const LO diff_j = j0 - j1;
+            const LO diff_k = j0 - k1;
+            
+            auto block = blocks(loc);
+            for (LO l0=0;l0<blocksize;++l0) 
+              for (LO l1=0;l1<blocksize;++l1) 
+                block(l0, l1) = get_block_crs_entry<value_type>(i0, j0, k0,
+                                                                diff_i, diff_j, diff_k,
+                                                                l0, l1);
+          }
+        });
+    }
+    {
+      RCP<TimeMonitor> timerBlockCrsFillComplete 
+        = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("3) BlockCrsMatrix FillComplete - currently do nothing")));
+      // this function does not exsit 
```

Comment:
"exist"

Reply:
also it doesn't exist on purpose (though we may want to rethink that at some point)